### PR TITLE
Clarifications/fixes and Ingress

### PR DESCRIPTION
**Pull request by kkohbrok**
This PR includes some editorial fixes/clarifications and changes the component that terminates TLS connections (and thus authenticates the sending party) to ingress instead of federator.

## Checklist:

Please tick the following before making your PR:

* [x] I ran `make` on this branch to build the docs and there are **no WARNINGs**.
* [x] After running `make`, I looked at the output by opening build/html/index.html in a browser and I'm satisfied with the rendering.

### CODE REVIEW

File: src/understand/federation/introduction.rst [hunk 1]
```diff
@@ -18,8 +18,9 @@ network.
 To enable federation, administrators of a Wire backend can decide to either
 specifically list the backends that they want to federate with, or to allow federation with all Wire backends reachable from the network.
 
-Federation is facilitated by a backend component called the *federator*, which
-acts as the ingress and egress point for federated communication.
+Federation is facilitated by two backend components: the *ingress*, which, as
+the name suggests, acts as ingress point for federated traffic and the
+*federator*, which acts as egress point.
```
Comment (jschaul) [inline src/understand/federation/introduction.rst:23]:
```suggestion
*federator*, which acts as egress point, as well as processing all ingress requests from the *ingress* after the authentication step.
```

File: src/understand/federation/architecture.rst [hunk 2]
```diff
@@ -78,7 +80,7 @@ while the domain that the backend is hosted at is called `infrastructure domain`
 
 To make discovery possible, any party hosting a Wire backend has to announce the
 the infra domain via a DNS `SRV` record as defined in `RFC 2782
-<https://tools.ietf.org/html/rfc2782>`_ with `service = wire-server-federator, proto =
+<https://tools.ietf.org/html/rfc2782>`_ with `service = wire-server-ingress, proto =
```
[the remark this answers is not in the document]
Comment (kkohbrok) [reply src/understand/federation/architecture.rst:83]: Ah, I see. I was under the impression that discovery, authentication and authorization were not implemented yet and thought that "ingress" was a more fitting name, as that's the component that actually gets discovered.

File: src/understand/federation/img/exact-handle-search.txt [hunk 3]
```diff
@@ -18,6 +18,8 @@ federator@B -> brig@B: (handle="alice")
 
 brig@B -> federator@B: Response: UserProfile of Alice
 
-federator@B -> federator@A: Response: UserProfile of Alice
+federator@B -> ingress@B: Response: UserProfile of Alice
+
+ingress@B -> federator@A: Response: UserProfile of Alice
```
Comment (jschaul) [inline src/understand/federation/img/exact-handle-search.txt:23]: Images are currently not automatically generated (unfortunately, we wanted to provide support for such a thing at some point, but will take some time to set up). In the meantime, could you copy this to https://swimlanes.io , export the png, and replace the exact-handle-search.png? I see that file was deleted.

**Review by jschaul — APPROVED** [review]
Minus the missing image, this now looks good!

File: src/understand/federation/architecture.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (jschaul) [reply src/understand/federation/architecture.rst]: Discovery is implemented actually. Just authentication/authorization is not yet.

File: src/understand/federation/img/exact-handle-search.txt (hunk 3 above)
Comment (kkohbrok) [reply src/understand/federation/img/exact-handle-search.txt:23]: Does it show as deleted? I'm pretty sure I committed a new version of that file. On my GH "files changed" view it just says that it's now 108 KB smaller (for whatever reason). Can you check again?

Comment (jschaul) [reply src/understand/federation/img/exact-handle-search.txt:23]: Ah, nevermind then. Sorry, I saw something red and thought the file had been removed. Sorry!

**Pull request merged**